services/py-api/src/service/participants_registration_service.py

```diff
@@ -42,3 +43,15 @@ async def register_random_participant(self, input_data: ParticipantRequestBody)
 
         # Proceed with registration if there is capacity
         return await self._hackathon_service.create_random_participant(input_data)
+    
+    async def register_invite_link_participant(self, input_data: ParticipantRequestBody) -> Result[
+        Tuple[Participant, Team],
+        DuplicateEmailError | DuplicateTeamNameError | HackathonCapacityExceededError | Exception,
+    ]:
+        # Check Team Capacity
+        has_capacity = await self._hackathon_service.check_team_capacity(input_data.team_name)
+        if not has_capacity:
+            return Err(TeamCapacityExceededError())
```

Comment:
Suggestion: you can do the decoding of the jwt here and move the rest of the logic in the hackathon service.

Reply:
Fixed in commit d8f0832f77647d818b784c5ad0942048bd62a834